source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -490,7 +500,7 @@ void DnsCacheImpl::finishResolve(const std::string& host,
     // We only set details here if current address is null because but
     // non-null->null resolutions we don't update the address so will use a
     // previously resolved address + details.
-    primary_host_info->host_info_->setDetails(std::string(details));
+    primary_host_info->host_info_->setDetails(details_with_maybe_trace);
```

Comment:
use std::move on the string?

Reply:
We may also want to change the setDetails() signature from `setDetails(std::string details)` to `setDetails(std::string&& details)` to force the move call.  How it stands now, it's easy to create multiple copies of the string when calling setDetails